sympy/solvers/ode/tests/test_ode.py

```diff
@@ -1458,10 +1342,9 @@ def test_solve_ics():
 
     # Test cases where dsolve returns two solutions.
     eq = (x**2*f(x)**2 - x).diff(x)
-    assert dsolve(eq, f(x), ics={f(1): 0}) == [Eq(f(x),
-        -sqrt(x - 1)/x), Eq(f(x), sqrt(x - 1)/x)]
-    assert dsolve(eq, f(x), ics={f(x).diff(x).subs(x, 1): 0}) == [Eq(f(x),
-        -sqrt(x - S.Half)/x), Eq(f(x), sqrt(x - S.Half)/x)]
+    sol = [Eq(f(x), -sqrt(C1 + x)/x), Eq(f(x), sqrt(C1 + x)/x)]
+    assert dsolve(eq) == sol
```

Comment:
The previous tests here used initial conditions. Why are they changed?

Reply:
@oscarbenjamin I think there was misunderstanding with this test case that I encountered, I will revoke this right away.